### PR TITLE
Fixes #13503: Add support to private repository to rtf

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/13503

### CODE REVIEW

**Comment by peckpeck** [comment]
you need to rebase

**Comment by Fdall** [comment]
PR rebased

File: scenario/ncf_tests.py [hunk 1]
```diff
@@ -15,13 +16,22 @@
...
+  if (download_user and download_password):
+    export_prefix = "export DOWNLOAD_USER=" + download_user + " DOWNLOAD_PASSWORD=" + download_password
+except:
+  export_prefix = ""
 
 # Get setup_ncf
 shell_on("agent", "wget -O /tmp/ncf-setup https://www.rudder-project.org/tools/ncf-setup", live_output=True)
 # Call setup_ncf
 if not cfengine_version:
   shell_on("agent", "sh /tmp/ncf-setup test-local https://github.com/Normation/ncf.git#branches/rudder/" + tag + " rudder-" + tag, live_output=True)
```
Comment (peckpeck) [inline scenario/ncf_tests.py]: if not cfengine version, means we use rudder, no ?
So we neeed it here but not below

File: vagrant.rb [hunk 2]
```diff
@@ -179,6 +179,7 @@ def provisioning_script(os, host_name, net, first_ip,
       command += "#{proxy} ALLOWEDNETWORK=#{net}.0/24 UNSUPPORTED=#{ENV['UNSUPPORTED']} REPO_PREFIX=rtf/ /usr/local/bin/rudder-setup setup-#{setup} \"#{version}\" \"#{server}\"\n"
     end
     if setup == "ncf" then
+      command += "export DOWNLOAD_USER=#{ENV['DOWNLOAD_USER']} DOWNLOAD_PASSWORD=#{ENV['DOWNLOAD_PASSWORD']}\n"
```
Comment (peckpeck) [inline vagrant.rb]: This is not specific to ncf

**Comment by Fdall** [comment]
Commit modified

File: scenario/ncf_tests.py [hunk 3]
```diff
@@ -15,13 +16,22 @@
 tag = get_param("tag", "")
 # Get CFEngine version used for the tests
 cfengine_version = get_param("cfengine_version", "")
+export_prefix = ""
+try:
+  download_user = os.environ['DOWNLOAD_USER']
+  download_password = os.environ['DOWNLOAD_PASSWORD']
+
+  if (download_user and download_password):
```
Comment (peckpeck) [inline scenario/ncf_tests.py]: what about empty password ?

File: scenario/ncf_tests.py [hunk 4]
```diff
@@ -15,13 +16,22 @@
...
 
 # Get setup_ncf
 shell_on("agent", "wget -O /tmp/ncf-setup https://www.rudder-project.org/tools/ncf-setup", live_output=True)
 # Call setup_ncf
 if not cfengine_version:
-  shell_on("agent", "sh /tmp/ncf-setup test-local https://github.com/Normation/ncf.git#branches/rudder/" + tag + " rudder-" + tag, live_output=True)
+  shell_on("agent", export_prefix + ";sh /tmp/ncf-setup test-local https://github.com/Normation/ncf.git#branches/rudder/" + tag + " rudder-" + tag, live_output=True)
 else:
-  shell_on("agent", "sh /tmp/ncf-setup test-local https://github.com/Normation/ncf.git#branches/rudder/" + tag + " " + cfengine_version, live_output=True)
+  shell_on("agent", export_prefix + ";sh /tmp/ncf-setup test-local https://github.com/Normation/ncf.git#branches/rudder/" + tag + " " + cfengine_version, live_output=True)
```
Comment (peckpeck) [inline scenario/ncf_tests.py]: do we need it when we install cfengine and not rudder ?

Comment (Fdall) [reply scenario/ncf_tests.py]: Yes, since cfengine_version can be designed like "rudder-latest" which will install the Rudder agent, and use its cf-agent.

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
> what about empty password ?
It should be ok now, empty user and non existent env  var should result in empty prefix.
Empty password will be ok

**Pull request merged**